src/apis/price-updater.ts

```diff
@@ -1,20 +1,43 @@
 import axios from "axios";
+import * as z from "zod";
 
 import { CurrencySymbol } from "src/utils/currencies";
 
 // domain
 import { FiatExchangeRates, Token } from "src/domain/hermez";
 
+// persistence
+import * as parsers from "src/persistence/parsers";
+
+const tokenListParser: z.ZodSchema<Token[]> = z.array(parsers.token);
+
+interface GetTokensPriceResponse {
+  tokens: Token[];
+}
+
+const getTokensPriceResponseParser: z.ZodSchema<GetTokensPriceResponse> = z.object({
+  tokens: tokenListParser,
+});
+
+// api exchange rates validation
 interface ApiExchangeRate {
   currency: string;
   price: number;
 }
 
-const baseApiUrl = `${process.env.REACT_APP_PRICE_UPDATER_API_URL}`;
-const apiKey = `${process.env.REACT_APP_PRICE_UPDATER_API_KEY}`;
-const client = axios.create({
-  baseURL: baseApiUrl,
-  headers: { "X-API-KEY": apiKey },
+const apiExchangeRateParser: z.ZodSchema<ApiExchangeRate> = z.object({
+  currency: z.string(),
+  price: z.number(),
+});
+
+const apiExchangeRateListParser: z.ZodSchema<ApiExchangeRate[]> = z.array(apiExchangeRateParser);
+
+interface ApiExchangeRateResponse {
+  currencies: ApiExchangeRate[];
+}
+
+const getFiatExchangeRatesResponseParser: z.ZodSchema<ApiExchangeRateResponse> = z.object({
```

Comment:
Same as before, shouldn't we move all this parsers and interfaces to persistence and the domain modules?

Reply:
The same answer applies too.